src/main/java/com/microsoft/jenkins/containeragents/builders/AciDeploymentTemplateBuilder.java

```diff
@@ -0,0 +1,351 @@
+
+
+        ArrayNode resourcesArray = (ArrayNode) tmp.get("resources");
+
+        ObjectNode containerGroupItem = (ObjectNode) resourcesArray.get(0);
+        ObjectNode propertiesNode = (ObjectNode) containerGroupItem.get("properties");
+        propertiesNode.set("networkProfile", networkProfileNode);
+        ArrayNode dependencyArrayNode = mapper.createArrayNode();
+        dependencyArrayNode.add("[resourceId('Microsoft.Network/networkProfiles', variables('containerName'))]");
+        containerGroupItem.set("dependsOn", dependencyArrayNode);
+
+        try (InputStream networkProfileSnippetStream =
+                     AciService.class.getResourceAsStream(NETWORK_PROFILE_SNIPPET_FILENAME)) {
+            JsonNode networkProfileItem = mapper.readTree(networkProfileSnippetStream);
+            resourcesArray.add(networkProfileItem);
+        }
+    }
+
+    private String mapIpType(AciPrivateIpAddress privateIpAddress) {
+        return privateIpAddress != null ? "Private" : "Public";
+    }
+
+    private void addPortNode(JsonNode tmp, ObjectMapper mapper, String port) {
+        JsonNode propertiesNode = tmp.get("resources").get(0).get("properties");
+        ArrayNode containerPortsNodes = (ArrayNode) propertiesNode.get("containers")
+                .get(0).get("properties").get("ports");
+        ArrayNode ipPortsNodes = (ArrayNode) propertiesNode.get("ipAddress").get("ports");
+
+        ObjectNode newContainerPortNode = mapper.createObjectNode();
+        newContainerPortNode.put("port", port);
+        containerPortsNodes.add(newContainerPortNode);
+
+        ObjectNode newIpPortNode = mapper.createObjectNode();
+        newIpPortNode.put("protocol", "tcp");
+        newIpPortNode.put("port", port);
+        ipPortsNodes.add(newIpPortNode);
+    }
+
+    private void addCommandNode(JsonNode tmp, String[] commands) {
+        ArrayNode commandNode = (ArrayNode) tmp.get("resources").get(0)
+                .get("properties").get("containers").get(0)
+                .get("properties").get("command");
+
+        for (String command : commands) {
+            commandNode.add(command);
+        }
+    }
+
+    private void addCommandNode(JsonNode tmp, String command, AciAgent agent) {
+        if (StringUtils.isBlank(command)) {
+            return;
+        }
+        String replaceCommand = commandReplace(command, agent);
+        addCommandNode(tmp, StringUtils.split(replaceCommand, ' '));
+    }
+
+    private void addLogAnalytics(JsonNode tmp, ObjectNode parameters,
+                                        ObjectMapper mapper, AciCloud aciCloud) {
+
+        if (StringUtils.isBlank(aciCloud.getLogAnalyticsCredentialsId())) {
+            return;
+        }
+
+        StandardUsernamePasswordCredentials credentials = CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StandardUsernamePasswordCredentials.class,
+                        jenkinsHelper.getJenkins(),
```

Comment:
you can pass null here, see https://github.com/jenkinsci/warnings-ng-plugin/blob/9c98e8e659fc17c4edcf4087219b92495e482f59/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/axivion/AxivionSuite.java#L196

```suggestion
                        (Item) null,
```